src/Plugin/StrawberryRunnersPostProcessor/OcrPostProcessor.php

```diff
@@ -0,0 +1,350 @@
+
+    $command = '';
+    $can_run_gs = \Drupal::service('strawberryfield.utility')->verifyCommand($execpath_gs);
+    $can_run_tesseract = \Drupal::service('strawberryfield.utility')->verifyCommand($execpath_tesseract);
+    $filename = pathinfo($file_path, PATHINFO_FILENAME);
+    $sourcefolder =  pathinfo($file_path,PATHINFO_DIRNAME);
+    $sourcefolder = strlen($sourcefolder)> 0 ? $sourcefolder.'/' : sys_get_temp_dir().'/';
+    $gs_destination_filename = "{$sourcefolder}{$filename}_{$page_number}.png";
+    if ($can_run_gs &&
+      $can_run_tesseract &&
+      (strpos($arguments_gs, '%file' ) !== FALSE) &&
+      (strpos($arguments_tesseract, '%file' ) !== FALSE)) {
+      $arguments_gs = "-dBATCH -dNOPAUSE -r300 -dUseCropBox -dQUIET -sDEVICE=pnggray -dFirstPage={$page_number} -dLastPage={$page_number} -sOutputFile=$gs_destination_filename " . $arguments_gs;
+      $arguments_gs = str_replace('%s','', $arguments_gs);
+      $arguments_gs = str_replace_first('%file','%s', $arguments_gs);
+      $arguments_gs = sprintf($arguments_gs, $file_path);
+
+      $arguments_tesseract = str_replace('%s','', $arguments_tesseract);
+      $arguments_tesseract = str_replace_first('%file','%s', $arguments_tesseract);
+      $arguments_tesseract = sprintf($arguments_tesseract, $gs_destination_filename);
+
+      $command_gs = escapeshellcmd($execpath_gs.' '.$arguments_gs);
+      $command_tesseract = escapeshellcmd($execpath_tesseract.' '.$arguments_tesseract);
+
+      $command = $command_gs.' && '.$command_tesseract;
+
+    } else {
+      error_log("missing arguments for OCR");
+    }
+    // Only return $command if it contains the original filepath somewhere
+    if (strpos($command, $file_path) !== false) { return $command;}
+    return '';
+
+  }
+
+  protected function hOCRtoMiniOCR($output, $pageid) {
+    error_log($output);
+    $hocr = simplexml_load_string($output);
+    $internalErrors = libxml_use_internal_errors(TRUE);
+    libxml_clear_errors();
+    libxml_use_internal_errors($internalErrors);
+    if (!$hocr) {
+      error_log('Could not convert HOCR to MiniOCR, sources is not valid XML');
+      return NULL;
+    }
+    $w = new \XMLWriter();
+    $w->openMemory();
+    $w->startDocument('1.0','UTF-8');
+    $w->startElement("ocr");
+    foreach ($hocr->body->children() as $page) {
+      $coos = explode(" ", substr($page['title'], 5));
+      if (count($coos)) {
+        $w->startElement("p");
+        $w->writeAttribute("id", $pageid);
+        $w->writeAttribute("wh", $coos[2] . " " . $coos[3]);
+        $w->startElement("b");
+        foreach ($page->children() as $line) {
+          $w->startElement("l");
+          foreach ($line->children() as $word) {
+            $wcoos = explode(" ", $word['title']);
+            if (count($wcoos)) {
+              $w->startElement("w");
+              $w->writeAttribute("x", $wcoos[1] . ' ' . $wcoos[2] . ' ' . $wcoos[3] . ' ' . $wcoos[4]);
```

Comment:
Instead of $wcoos[3] we have to put $wcoos[3]-$wcoos[1]
Instead of $wcoos[4] we have to put $wcoos[4]-$wcoos[2]

Reply:
Ok will do that